core/src/autogluon/core/models/ensemble/fold_fitting_strategy.py

```diff
@@ -726,10 +752,27 @@ def _run_pseudo_sequential(self, X, y, X_pseudo, y_pseudo, model_base_ref, time_
 
         self._update_bagged_ensemble_times()
 
+    def _calculate_gpu_assignment(self, gpu_assignments: Dict, task_id: int, gpus_per_task: int, total_gpus: int):
+        if total_gpus == 0:
+            logger.debug(f"No GPUs available, CPU-only mode for task {task_id}")
+            gpu_assignments[task_id] = []
+            return gpu_assignments   
+        if gpus_per_task >= 1:
+            gpu_id = task_id * gpus_per_task
+            assigned_gpus = []
+            for i in range(gpus_per_task):
+                assigned_gpus.append(gpu_id + i % total_gpus)
+            gpu_assignments[task_id] = assigned_gpus
+        else:
+            gpu_id = task_id % total_gpus
+            gpu_assignments[task_id] = [gpu_id]
+        return gpu_assignments
```

Comment:
Why are we doing in-place edits to gpu_assignments?
Why not just return the value we want to set it to, and then

```
gpu_assignments[task_id] = self._calculate_gpu_assignment(task_id, gpus_per_task, total_gpus)
```